pyccel/codegen/printing/ccode.py

```diff
@@ -268,6 +393,84 @@ def function_signature(self, expr):
             arg_code = ', '.join('{0}{1}'.format(self.get_declare_type(i), i) for i in expr.arguments)
         return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_NumpyUfuncBase(self, expr):
+        """ convert a python expresion with a numpy function call to c
+        function call
+
+        Parameters:
+        -----------
+            expr (Pyccel ast node): python expresion with a numpy function call
+
+        Returns:
+        --------
+            string: equivalent of the expression in c language
+
+        Example:
+        -------
+            numpy.cos(x) ==> cos(x)
+
+        """
+        # add necessary include
+        self._additional_imports.add('math.h')
+        type_name = type(expr).__name__
+        func_name = numpy_ufunc_to_c_real[type_name]
+        args = []
+        for arg in expr.args:
+            if arg.dtype is NativeComplex():
+                self._additional_imports.add('complex.h')
+                try:
+                    func_name = numpy_ufunc_to_c_complex[type_name]
+                    args.append(self._print(arg))
+                except KeyError:
+                    errors.report(INCOMPATIBLE_TYPEVAR_TO_FUNC.format(type_name) ,severity='fatal')
+            elif arg.dtype is not NativeReal():
+                args.append(self._print(NumpyFloat(arg)))
+            else :
+                args.append(self._print(arg))
+        code_args = ', '.join(args)
+        return '{0}({1})'.format(func_name, code_args)
+
+    def _print_MathFunctionBase(self, expr):
+        """ convert a python expresion with a math function call to c
+        function call
+
+        Parameters:
+        -----------
+            expr (Pyccel ast node): python expresion with a Math function call
+
+        Returns:
+        --------
+            string: equivalent of the expression in c language
+
+        Example:
+        --------
+            math.sin(x) ==> sin(x)
+
+        """
+        # add necessary include
+        self._additional_imports.add('math.h')
+        type_name = type(expr).__name__
+        func_name = math_function_to_c[type_name]
```

Comment:
Please add a try/except statement to catch possible `KeyError`s if the user requests a function which is not implemented. This can be used to raise a neat `PYCCEL_RESTRICTION_TODO`

Reply:
done